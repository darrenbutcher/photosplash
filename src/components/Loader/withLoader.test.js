import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { withLoader } from './index'

describe('withLoader', () => {
  const App = (props) => (<div>Content</div>)
  const AppWithLoading = withLoader(App)
  const wrapper = shallow(<AppWithLoading isFetching={false} />)
  const isFetchingWrapper = shallow(<AppWithLoading isFetching={true} />)

  it('should render loading message when isFetching', () => {
    expect(toJson(isFetchingWrapper)).toMatchSnapshot()
  })

  it('should render content when isFetching is false', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
