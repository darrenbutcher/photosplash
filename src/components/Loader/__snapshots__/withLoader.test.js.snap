// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoader should render content when isFetching is false 1`] = `
<App
  isFetching={false}
/>
`;

exports[`withLoader should render loading message when isFetching 1`] = `
<Dimmer
  active={true}
  inverted={true}
>
  <Loader
    inverted={true}
  />
</Dimmer>
`;
